Browser tests appear to be running with non-browser code

I attempted to refactor the tests to use mocha "tags" to prevent non-browser tests from running in the browser (i.e. instead of using `isBrowser` or `process.browser`).  See https://github.com/tekwiz/async/commit/176b947043500767818c3b67396dfd9fb26260af.  However, when I did this, the browser tests failed to run:

```
START:
12 07 2016 12:51:02.153:INFO [framework.browserify]: bundle built
12 07 2016 12:51:02.169:INFO [karma]: Karma v0.13.22 server started at http://localhost:9876/
12 07 2016 12:51:02.176:INFO [launcher]: Starting browser Firefox
12 07 2016 12:51:04.238:INFO [Firefox 45.0.0 (Mac OS X 10.11.0)]: Connected on socket /#IAMzwTa1Tv27JMNsAAAA with id 45738836
Firefox 45.0.0 (Mac OS X 10.11.0) ERROR
  Error: Cannot find module 'native-promise-only'
  at /var/folders/yl/pg2l6p6j5nb9twhsp8ymqktc0000gn/T/87c1e0334e1cc54f87ea66cc4683870f.browserify:1 <- node_modules/browser-pack/_prelude.js:1:0

Finished in 0.425 secs / 0 secs

SUMMARY:
✔ 0 tests completed
```

I was able to narrow down the problem to `mocha_test/asyncify.js` -- it occurs when the asyncify - promisified suite is not loaded.  My theory is that browserify or browser-pack is attempting to load native-promise-only because it is referenced but the other is not loading it because it's never actually run.

I'm not really sure how go about debugging this, so any pointers would be appreciated.

--- a/karma.conf.js
+++ b/karma.conf.js
@@ -12,6 +12,13 @@ module.exports = function(config) {
         browserify: {
             debug: true,
             transform: ['babelify']
+        },
+
+        client: {
+            mocha: {
+                grep: '@nodeonly',
+                invert: true
+            }
         }
     });
 };

--- a/mocha_test/asyncify.js
+++ b/mocha_test/asyncify.js
@@ -1,7 +1,6 @@
 var async = require('../lib');
 var assert = require('assert');
 var expect = require('chai').expect;
-var isBrowser = require('./support/is_browser');
 
 describe('asyncify', function(){
 
@@ -61,12 +60,7 @@ describe('asyncify', function(){
         }
     });
 
-    describe('promisified', function() {
-        if (isBrowser()) {
-            // node only tests
-            return;
-        }
-
+    describe('promisified @nodeonly', function() {
         var names = [
             'native-promise-only',
             'bluebird',

--- a/mocha_test/asyncify.js
+++ b/mocha_test/asyncify.js
@@ -60,84 +60,79 @@ describe('asyncify', function(){
         }
     });
 
-    describe('promisified @nodeonly', function() {
-        var names = [
-            'native-promise-only',
-            'bluebird',
-            'es6-promise',
-            'rsvp'
-        ];
-
-        // Both Bluebird and native promises emit these events. We handle it because Bluebird
-        // will report these rejections to stderr if we don't, which is a great feature for
-        // normal cases, but not here, since we expect unhandled rejections:
-        function ignoreRejections() {}
-
-        before(function () {
-            process.on('unhandledRejection', ignoreRejections);
-        });
-
-        after(function () {
-            process.removeListener('unhandledRejection', ignoreRejections);
-        });
-
-        names.forEach(function(name) {
-            describe(name, function() {
-
-                var Promise = require(name);
-                if (typeof Promise.Promise === 'function') {
-                    Promise = Promise.Promise;
-                }
-
-                it('resolve', function(done) {
-                    var promisified = function(argument) {
-                        return new Promise(function (resolve) {
-                            setTimeout(function () {
-                                resolve(argument + " resolved");
-                            }, 15);
-                        });
-                    };
-                    async.asyncify(promisified)("argument", function (err, value) {
-                        if (err) {
-                            return done(new Error("should not get an error here"));
-                        }
-                        expect(value).to.equal("argument resolved");
-                        done();
-                    });
-                });
-
-                it('reject', function(done) {
-                    var promisified = function(argument) {
-                        return new Promise(function (resolve, reject) {
-                            reject(argument + " rejected");
-                        });
-                    };
-                    async.asyncify(promisified)("argument", function (err) {
-                        assert(err);
-                        expect(err.message).to.equal("argument rejected");
-                        done();
-                    });
-                });
-
-                it('callback error', function(done) {
-                    var promisified = function(argument) {
-                        return new Promise(function (resolve) {
+    describe('promisified', function() {
+        function promisifiedTests(Promise) {
+            it('resolve', function(done) {
+                var promisified = function(argument) {
+                    return new Promise(function (resolve) {
+                        setTimeout(function () {
                             resolve(argument + " resolved");
-                        });
-                    };
-                    var call_count = 0;
-                    async.asyncify(promisified)("argument", function () {
-                        call_count++;
-                        if (call_count === 1) {
-                            throw new Error("error in callback");
-                        }
+                        }, 15);
                     });
-                    setTimeout(function () {
-                        expect(call_count).to.equal(1);
-                        done();
-                    }, 15);
+                };
+                async.asyncify(promisified)("argument", function (err, value) {
+                    if (err) {
+                        return done(new Error("should not get an error here"));
+                    }
+                    expect(value).to.equal("argument resolved");
+                    done();
                 });
             });
+
+            it('reject', function(done) {
+                var promisified = function(argument) {
+                    return new Promise(function (resolve, reject) {
+                        reject(argument + " rejected");
+                    });
+                };
+                async.asyncify(promisified)("argument", function (err) {
+                    assert(err);
+                    expect(err.message).to.equal("argument rejected");
+                    done();
+                });
+            });
+
+            it('callback error', function(done) {
+                var promisified = function(argument) {
+                    return new Promise(function (resolve) {
+                        resolve(argument + " resolved");
+                    });
+                };
+                var call_count = 0;
+                async.asyncify(promisified)("argument", function () {
+                    call_count++;
+                    if (call_count === 1) {
+                        throw new Error("error in callback");
+                    }
+                });
+                setTimeout(function () {
+                    expect(call_count).to.equal(1);
+                    done();
+                }, 15);
+            });
+        }
+
+        describe('native-promise-only', function() {
+            var Promise = require('native-promise-only');
+            promisifiedTests.call(this, Promise);
+        });
+
+        describe('bluebird', function() {
+            var Promise = require('bluebird');
+            // Bluebird reports unhandled rejections to stderr. We handle it because we expect
+            // unhandled rejections:
+            Promise.onPossiblyUnhandledRejection(function ignoreRejections() {});
+            promisifiedTests.call(this, Promise);
+        });
+
+        describe('es6-promise', function() {
+            var Promise = require('es6-promise').Promise;
+            promisifiedTests.call(this, Promise);
+        });
+
+        describe('rsvp', function() {
+            var Promise = require('rsvp').Promise;
+            promisifiedTests.call(this, Promise);
         });
     });
 });

--- a/mocha_test/forever.js
+++ b/mocha_test/forever.js
@@ -1,6 +1,5 @@
 var async = require('../lib');
 var expect = require('chai').expect;
-var isBrowser = require('./support/is_browser');
 
 describe('forever', function(){
     context('function is asynchronous', function(){
@@ -24,8 +23,7 @@ describe('forever', function(){
     });
 
     context('function is synchronous', function(){
-        it('does not cause a stack overflow', function(done){
-            if (isBrowser()) return done(); // this will take forever in a browser
+        it('does not cause a stack overflow @nodeonly', function(done){ // this will take forever in a browser
             var counter = 0;
             function addOne(callback) {
                 counter++;

--- a/mocha_test/nextTick.js
+++ b/mocha_test/nextTick.js
@@ -13,12 +13,7 @@ describe("nextTick", function () {
         }, 50);
     });
 
-    it('nextTick in the browser', function(done){
-        if (!process.browser) {
-            // skip this test in node
-            return done();
-        }
-
+    it('nextTick in the browser @nodeonly', function(done){
         var call_order = [];
         async.nextTick(function(){call_order.push('two');});
 

--- a/mocha_test/parallel.js
+++ b/mocha_test/parallel.js
@@ -1,7 +1,6 @@
 var async = require('../lib');
 var expect = require('chai').expect;
 var assert = require('assert');
-var isBrowser = require('./support/is_browser');
 var getFunctionsObject = require('./support/get_function_object');
 
 describe('parallel', function() {
@@ -192,12 +191,7 @@ describe('parallel', function() {
         });
     });
 
-    it('parallel call in another context @nycinvalid', function(done) {
-        if (isBrowser()) {
-            // node only test
-            done();
-            return;
-        }
+    it('parallel call in another context @nycinvalid @nodeonly', function(done) {
         var vm = require('vm');
         var sandbox = {
             async: async,

--- a/mocha_test/series.js
+++ b/mocha_test/series.js
@@ -1,7 +1,6 @@
 var async = require('../lib');
 var expect = require('chai').expect;
 var assert = require('assert');
-var isBrowser = require('./support/is_browser');
 var getFunctionsObject = require('./support/get_function_object');
 
 describe('series', function() {
@@ -137,12 +136,7 @@ describe('series', function() {
         });
     });
 
-    it('call in another context @nycinvalid', function(done) {
-        if (isBrowser()) {
-            // node only test
-            done();
-            return;
-        }
+    it('call in another context @nycinvalid @nodeonly', function(done) {
         var vm = require('vm');
         var sandbox = {
             async: async,

--- a/mocha_test/support/is_browser.js
+++ b/mocha_test/support/is_browser.js
@@ -1,4 +1,0 @@
-module.exports = function() {
-    return (typeof process === "undefined") ||
-        (process + "" !== "[object process]"); // browserify
-};

--- a/mocha_test/waterfall.js
+++ b/mocha_test/waterfall.js
@@ -106,13 +106,7 @@ describe("waterfall", function () {
         }).to.throw(/already called/);
     });
 
-    it('call in another context @nycinvalid', function(done) {
-        if (process.browser) {
-            // node only test
-            done();
-            return;
-        }
-
+    it('call in another context @nycinvalid @nodeonly', function(done) {
         var vm = require('vm');
         var sandbox = {
             async: async,

